discord/src/main.rs

```diff
@@ -0,0 +1,314 @@
+use std::env;
+use std::time::{Duration};
+use std::net::SocketAddr;
+use serde::{Serialize, Deserialize};
+use serde_json::Value;
+use lazy_static::lazy_static;
+use openssl::rsa::{Rsa, Padding};
+use reqwest::{Client, ClientBuilder};
+use axum::{
+	Router,
+	routing::{get, post},
+	extract::{Query, Json},
+	response::{IntoResponse},
+	http::{StatusCode},
+};
+
+lazy_static! {
+	static ref REACTOR_API_PREFIX: String = env::var("REACTOR_API_PREFIX").expect("Env variable REACTOR_API_PREFIX not set");
+	static ref REACTOR_AUTH_TOKEN: String = env::var("REACTOR_AUTH_TOKEN").expect("Env variable REACTOR_AUTH_TOKEN not set");
+
+	static ref DISCORD_APP_CLIENT_ID: String = env::var("DISCORD_APP_CLIENT_ID").expect("Env variable DISCORD_APP_CLIENT_ID not set");
+	static ref DISCORD_APP_CLIENT_SECRET: String = env::var("DISCORD_APP_CLIENT_SECRET").expect("Env variable DISCORD_APP_CLIENT_SECRET not set");
+	static ref DISCORD_REDIRECT_URL: String = env::var("DISCORD_REDIRECT_URL").expect("Env variable DISCORD_REDIRECT_URL not set");
+
+	static ref PASSPHRASE: String = env::var("PASSPHRASE").expect("Env variable PASSPHRASE not set");
+	static ref PUBLIC_KEY_PEM: String = env::var("PUBLIC_KEY_PEM").expect("Env variable PUBLIC_KEY_PEM not set");
+	static ref PRIVATE_KEY_PEM: String = env::var("PRIVATE_KEY_PEM").expect("Env variable PRIVATE_KEY_PEM not set");
+
+	static ref BOT_TOKEN: String = env::var("BOT_TOKEN").expect("Env variable BOT_TOKEN not set");
+
+}
+
+const TIMEOUT: u64 = 120;
+
+
+fn new_http_client() -> Client {
+	let cb = ClientBuilder::new().timeout(Duration::from_secs(TIMEOUT));
+	return cb.build().unwrap();
+}
+
+fn encrypt(data: &str) -> String {
```

Comment:
We found that using `Rsa` makes encryption easier and we hope to use it in the latest code. You can refer to https://github.com/second-state/reactor-connector-rs/blob/62fef756065827f6de7edfcfc3b093635ba42f70/github/src/main.rs#L118